### PR TITLE
Bedrock 1.20.10 protocol support

### PR DESCRIPTION
**Pull request by CreeperG16**
- Bumped verison to 1.20.10, protocol version to 594
- Changed `charged_crossbow` event ID to `charged_item`
- Updated `available_commands` packet to support `chained_subcommands`
- Ddded `missed_swing`, `start_crawling` and `stop_crawling` input flags and player actions
- added `agent_animation` packet (0x130)
- added `collision_box` metadata key, `crawling` metadata flag

### CODE REVIEW

File: data/bedrock/latest/proto.yml [hunk 1]
```diff
@@ -966,7 +966,7 @@ packet_entity_event:
       71: balloon_pop
       72: treasure_hunt
       73: agent_summon
-      74: charged_crossbow
+      74: charged_item
```
Comment (extremeheat) [inline data/bedrock/latest/proto.yml:969]: This doesn't seem related to a feature change, it seems like a rename for clarity. Let's either revert this or apply it to all the old versions too for consistency.

File: data/bedrock/latest/proto.yml [hunk 2]
```diff
@@ -3958,4 +3985,15 @@ packet_open_sign:
...
+# agent_animation is an Education Edition packet sent from the server to the client to make an agent perform an animation.
+packet_agent_animation:
+   !id: 0x130
+   !bound: client
+   # animation is the ID of the animation that the agent should perform. As of its implementation, there are no IDs
+   # that can be used in the regular client.
+   animation: u8
+   # entity_runtime_id is the runtime ID of the entity. The runtime ID is unique for each world session, and
+   # entities are generally identified in packets using this runtime ID.
+   entity_runtime_id: lu64
```
Comment (extremeheat) [inline data/bedrock/latest/proto.yml:3999]: this should be `varint64`

**Review by extremeheat — CHANGES_REQUESTED** [review]
Looks good overall, thanks. Left some comments on changes we should apply to the other versions also.

**Review by extremeheat — APPROVED** [review]

File: data/bedrock/latest/types.yml [hunk 3]
```diff
@@ -293,6 +293,10 @@ MetadataDictionary: []varint
       125: movement_sound_distance_offset
       126: heartbeat_interval_ticks
       127: heartbeat_sound_event
+      128: player_last_death_position
+      129: player_last_death_dimension
+      130: player_has_died
+      131: collision_box
```
[the remark this answers is not in the document]
Comment (extremeheat) [reply data/bedrock/latest/types.yml:299]: opened #735 to track the latter point

**Comment by extremeheat** [comment]
thanks!

**Comment by CreeperG16** [comment]
ah, forgot about that. That should be 1.20.10, not 1.12.10

**Comment by extremeheat** [comment]
yeah, readme typo will be fixed with #736
